### PR TITLE
Adding README

### PR DESCRIPTION
**Pull request by ineiti**
Adds a README with instructions how to build and use the directory directory.

See #39

### CODE REVIEW

**Comment by ineiti** [comment]
@tharvik please have a quick look that the README is somewhat understandable

File: tools/Makefile [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+directory-cache: pip
+	@if [[ ! -f authority_certificate ]]; then \
+	  echo "Please copy authority_* files to this directory"; \
+	  false; \
+	fi
+	mkdir -f directory
+	./gen_fresh_dirinfo.py
+	mv consensus.txt microdescriptors.txt directory
+
+pip:
```
Comment (tharvik) [inline tools/Makefile:12]: This target is an implicit `PHONY` as the file `pip` is never generated, so the dependencies will always get reinstalled

File: tools/README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+tor-nodes for faster circuit setup.
+You will have to let this run on a regular basis on a trusted server.
+Then the app will have to download the files, put them in a directory,
+and send the path to this directory to the library.
+
+Currently, you have to do the following:
+
+To create the relevant files, you need to do the following:
+
+1. Copy the `authority_``{certificate,identity_key,signing_key}` in the `tools/`-directory
```
Comment (tharvik) [inline tools/README.md:13]: How to generate theses files?

File: tools/README.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+2. run `make`
```
Comment (tharvik) [inline tools/README.md:14]:
```suggestion
2. run `make` in the `tools`-directory
```

File: tools/Makefile [hunk 4]
```diff
@@ -0,0 +1,14 @@
+all:	directory-cache
+
+directory-cache: pip
+	@if [[ ! -f authority_certificate ]]; then \
+	  echo "Please copy authority_* files to this directory"; \
+	  false; \
+	fi
```
[the remark this answers is not in the document]
Comment (ineiti) [reply tools/Makefile:7]: I thought about this, but couldn't get it to work... I'll try it again.

File: tools/Makefile [hunk 5]
```diff
@@ -0,0 +1,14 @@
+all:	directory-cache
+
+directory-cache: pip
```
[the remark this answers is not in the document]
Comment (ineiti) [reply tools/Makefile:3]: Fixing

File: tools/Makefile (hunk 1 above)
Comment (ineiti) [reply tools/Makefile:12]: Yes, any idea how to change that? Add a 'phony' file that will be created upon installation of the dependencies?

File: tools/README.md (hunk 2 above)
Comment (ineiti) [reply tools/README.md:13]: #39

File: README.md [hunk 6]
```diff
@@ -25,6 +25,16 @@ The asynchronous part needs to be done by the android and ios library.
 - `./` is the rust library for the wrapper with arti
 - `./ios` holds scripts to create the `XCFramework` used in the `arti-ios`
 
+# Pre-caching of tor circuits
+
+This library has a modified directory controller of arti that allows to use pre-downloaded
```
[the remark this answers is not in the document]
Comment (ineiti) [reply README.md:30]: s/directory controller/directory manager/

and added a link to the replacement files. Is that enough, or should I add more information?

File: tools/Makefile (hunk 4 above)
[the remark this answers is not in the document]
Comment (ineiti) [reply tools/Makefile:7]: OK, now it works. I tried it mentally, and it already gave an error. My brain-make is not compatible with gnu-make ;)

File: README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (tharvik) [reply README.md]: It's still not very clear what it is, why would a "directory" be needed for some network communication?

File: tools/Makefile (hunk 1 above)
Comment (tharvik) [reply tools/Makefile:12]: Yep, that's the way to do it :)

https://www.gnu.org/software/make/manual/html_node/Empty-Targets.html

File: README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (ineiti) [reply README.md]: I'm no specialist in Tor, but I guess for tor people this is quite clear. For example, there is a `tor-dirmgr` crate in arti: https://docs.rs/tor-dirmgr/0.0.0/tor_dirmgr/

File: tools/Makefile (hunk 1 above)
Comment (ineiti) [reply tools/Makefile]: OK, done...

**Pull request merged**